uber/site_sections/registration.py

```diff
@@ -97,7 +97,9 @@ def form(self, session, message='', return_to='', omit_badge='', check_in='', **
                 if check_in:
                     attendee.checked_in = localized_now()
                 session.add(attendee)
-                if return_to:
+                if params['badge_status'] == c.INVALID_STATUS:
+                    self.delete(attendee.id)
```

Comment:
Huh, does this work?  I guess it probably would.  In general we don't call page handlers from other page handlers, though there are some potential problems.  Most notably we're currently in the middle of one open session and calling the other other page handler opens the other session, which will commit before this one tries to commit.  At a glance that looks dangerous, so I'd be inclined to do this some other way, e.g. doing a redirect.  (Admittedly, doing a redirect is problematic for different reasons since the ``delete`` page handler currently expects a ``POST``.)